src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -393,7 +393,7 @@ function ( WP_HTML_Token $token ): void {
 				$provenance            = ( ! $same_node || $is_virtual ) ? 'virtual' : 'real';
 				$this->element_queue[] = new WP_HTML_Stack_Event( $token, WP_HTML_Stack_Event::PUSH, $provenance );
 
-				$this->change_parsing_namespace( $token->namespace );
+				$this->change_parsing_namespace( $token->integration_node_type ? 'html' : $token->namespace );
```

Comment:
Good spot, I think you're correct on this. I hadn't appreciated how integration_node_type works 👍